src/main/java/org/folio/services/finance/transaction/BaseTransactionService.java

```diff
@@ -74,63 +71,57 @@ private Future<TransactionCollection> getTransactionsChunk(List<String> transact
+    Batch batch = new Batch();
+    if (transactionsToCreate != null) {
+      transactionsToCreate.forEach(tr -> {
+        if (tr.getId() == null) {
+          tr.setId(UUID.randomUUID().toString());
+        }
+      });
+      batch.setTransactionsToCreate(transactionsToCreate);
+    }
+    if (transactionsToUpdate != null) {
+      batch.setTransactionsToUpdate(transactionsToUpdate);
+    }
+    if (idsOfTransactionsToDelete != null) {
+      batch.setIdsOfTransactionsToDelete(idsOfTransactionsToDelete);
+    }
+    if (transactionPatches != null) {
+      batch.setTransactionPatches(transactionPatches);
+    }
+    String endpoint = resourcesPath(FINANCE_BATCH_TRANSACTIONS);
+    return restClient.postEmptyResponse(endpoint, batch, requestContext)
+      .onSuccess(v -> logger.info("batchAllOrNothing completed successfully"))
+      .onFailure(t -> logger.error("batchAllOrNothing failed, batch={}", JsonObject.mapFrom(batch), t));
   }
 
-  public Future<Void> updateTransaction(Transaction transaction, RequestContext requestContext) {
-    return Optional.ofNullable(getByIdEndpoint(transaction))
-        .map(RequestEntry::new)
-        .map(requestEntry -> requestEntry.withId(transaction.getId()))
-        .map(requestEntry -> restClient.put(requestEntry, transaction, requestContext))
-        .orElseGet(this::unsupportedOperation);
+  public Future<Void> batchCreate(List<Transaction> transactions, RequestContext requestContext) {
+    return batchAllOrNothing(transactions, null, null, null, requestContext);
   }
 
-  public Future<Void> updateTransactions(List<Transaction> transactions, RequestContext requestContext) {
-    if (CollectionUtils.isEmpty(transactions)) {
-      return Future.succeededFuture();
-    }
-    return requestContext.getContext()
-      .<List<Future<Void>>>executeBlocking(promise -> {
-        List<Future<Void>> futures = new ArrayList<>();
-        var semaphore = new Semaphore(1, requestContext.getContext().owner());
-        for (Transaction tr : transactions) {
-          semaphore.acquire(() -> {
-            var future = updateTransaction(tr, requestContext)
-              .onComplete(asyncResult -> semaphore.release());
-
-            futures.add(future);
-            if (futures.size() == transactions.size()) {
-              promise.complete(futures);
-            }
-          });
-        }
-      })
-      .compose(GenericCompositeFuture::join)
-      .mapEmpty();
+  public Future<Void> batchUpdate(List<Transaction> transactions, RequestContext requestContext) {
+    return batchAllOrNothing(null, transactions, null, null, requestContext);
   }
 
-  public <T> Future<T> unsupportedOperation() {
-    Promise<T> promise = Promise.promise();
-    promise.fail(new UnsupportedOperationException());
-    return promise.future();
+  public Future<Void> batchRelease(List<Transaction> transactions, RequestContext requestContext) {
+    // NOTE: we will have to use transactionPatches when it is available
```

Comment:
WDYT about adding link to ticket if it exists?

Reply:
Sure